### ISSUE TITLE
Possibility to run original init executable

### ISSUE THREAD

**Issue by Hiller** [topjohnwu/Magisk#4272]
I have a problem installing Magisk. I have an Android TV box. I patched boot image and flashed it successfully. But when i'm trying to boot i'm getting a bootloop. After some investigations i found that restoring original init executable (repack ramdisk) fix the boot. I tried to run original init through custom.rc file using sh script but it looks like i got a "soft" loop or it just hung.
Why Magisk is deleting original init executable from ramdisk instead of running it after main magisk routines or vice versa?

May be there are already some solutions that can be used to run both init executables but i didn't find them. Boot images attached
[boot-resources.zip](https://github.com/topjohnwu/Magisk/files/6364949/boot-resources.zip)

Device:
Android version: Andriod TV 8
Magisk version name: 20.4-22.1
Magisk version code: 22104

**Comment by osm0sis** [comment]
Magisk does run the original init, your issue is more that there's something else the device expects from the original init which isn't being done or isn't being done at the correct time.

**Comment by Hiller** [comment]
if you can explain me how to save logs from kmsg to sdcard in such situation i can provide them.
I do not have fastboot mode on the device due to usb 3.0 on a board.  I tried Log-Catcher module but it doesn't work - i believe because it doesn't even started. 
And again: adding original init to custom.rc is leading to "soft" loop.

**Comment by osm0sis** [comment]
The best I can do is point you here: https://www.didgeridoohan.com/magisk/MagiskHelp

**Comment by Hiller** [comment]
Here's the boot log. 
[boot.log](https://github.com/topjohnwu/Magisk/files/6369302/boot.log)
Problem is here
`[    8.368674@3] init: cannot setexeccon('u:r:magisk:s0') for hgTbhGVQDAj: Invalid argument
[    8.373374@3] init: panic: rebooting to bootloader
`
Seems like no appropriate sepolicy configuration or context is missing.

**Comment by osm0sis** [comment]
Well more accurately I think it means your device didn't like the eepolicy change.

**Comment by Hiller** [comment]
Can it be related to magiskinit not beeing able to mount partitions. 

> [    5.073171@1] magiskinit: mount ->/vendor failed with 22: Invalid argument
[    5.115050@1] magiskinit: mount ->/system failed with 22: Invalid argument
[    5.156949@1] magiskinit: mount ->/odm failed with 22: Invalid argument

and finally not adding sepolicy rules?

**Comment by osm0sis** [comment]
Yep, possibly. Someone with experience with the magiskinit code will have to take a look. Thanks for the logs.

**Comment by vvb2060** [comment]
You are not using version 22104, we need debug log.

**Comment by Hiller** [comment]
I'm sorry, i flashed wrong image. Here's backup log
[boot_debug.log](https://github.com/topjohnwu/Magisk/files/6371251/boot_debug.log)

**Comment by Hiller** [comment]
Another boot log. This time i forced MagiskInit to use 2 stages (as in log file you can see original init has 2 stages). Here's log
[boot_debug.new.log](https://github.com/topjohnwu/Magisk/files/6371904/boot_debug.new.log)
This time problem is due to magisk init patched original init and IsDtFstabCompatible returned false, so original init refused to mount system

**Comment by vvb2060** [comment]
try it? [app-debug.zip](https://github.com/topjohnwu/Magisk/files/6374114/app-debug.zip)

**Comment by Hiller** [comment]
same
RootFSInit failed with no policy error and for 2 stages original init is still patching:

> [    4.923422@2] magiskinit: Replace [android,fstab] -> [xxx]
[    4.946376@2] magiskinit: Replace [android,fstab] -> [xxx]
[    4.951632@2] magiskinit: Write fstab file: fstab.amlogic
[    4.965692@2] magiskinit: Unmount [/sys]
[    4.989503@3] magiskinit: Unmount [/proc]
[    4.991684@3] init: init first stage started!
[    4.992400@3] init: Firstastage mount skipped (missing/incompatible fstab in device uree)

**Comment by vvb2060** [comment]
kmsg

**Comment by Hiller** [comment]
[Uploading boot.newest.log…]()

**Comment by vvb2060** [comment]
？
magiskinit: FirstStageInit

**Comment by Hiller** [comment]
I have 2 stages init.

> [    5.323163@0] init: init first stage started!
[    5.414215@3] init: Loading SELinux policy
[    5.590023@1] init: init second stage started!

So i added /apex folder to force magisk use it. 
According to logs and Magiskinit source code i think it won't work with RootFsInit. The problem with FirstStageInit is a patching 
of IsDtFstabCompatible  return in original init. So i believe Original init has smth like
`if(!IsDtFstabCompatible()) {
LogError("First stage mount skipped (missing/incompatible fstab in device tree)");
return;
}`

**Comment by Hiller** [comment]
Is it possible to get a magiskinit version without IsDtFstabCompatible patching?
